server/src/main/java/io/crate/execution/ddl/tables/AddColumnTask.java

```diff
@@ -123,34 +113,56 @@ public ClusterState execute(ClusterState currentState, AddColumnRequest request)
         return currentState;
     }
 
-    static AddColumnRequest addMissingParentColumns(AddColumnRequest request, DocTableInfo currentTable) {
+
+    /**
+     * Adds missing parent columns.
+     * Also tries to update request by taking existing references from the cluster state.
+     * @return NULL if all columns already exist.
+     */
+    @Nullable
+    static AddColumnRequest normalizeColumns(AddColumnRequest request, DocTableInfo currentTable) {
         List<Reference> newColumns = new ArrayList<>();
-        boolean anyMissing = false;
-        for (var newColumn : request.references()) {
-            if (newColumn.column().isTopLevel()) {
-                newColumns.add(newColumn);
+
+        // Get existing references
+        boolean allExist = true;
+        for (Reference ref : request.references()) {
+            Reference exists = currentTable.getReference(ref.column());
+            if (exists == null) {
+                allExist = false;
+                newColumns.add(ref); // Actually new column, take it from request.
```

Comment:
As mentioned in the [comment](https://github.com/crate/crate/pull/13757#discussion_r1127951615) to the old PR, we need to refresh references. We also had a [discussion](https://github.com/crate/crate/pull/13774#discussion_r1129235980) in another PR that we can combine refresh+parents together in smth like `normalizeColumns` - doing it now.